experimental/dds/tree2/src/index.ts

```diff
@@ -278,6 +278,15 @@ export {
 	Unenforced,
 	schemaIsFieldNode,
 	AllowedTypeSet,
+	SharedTreeList,
+	ObjectFields,
+	ProxyField,
```

Comment:
Do you like these "Proxy*" names? I just sort of chose them arbitrarily to not conflict with the existing similar ones and I think they're a little confusing. What do you think about `SharedTreeField`, `SharedTreeNode`, and `SharedTreeNodeUnion` to match the other types, at least for now? Ideally we wouldn't need the SharedTree prefix for any of them but `Node` and `Object` are too easy to conflict with.

Reply:
Yeah...  I think 'Proxy*' is good placeholder name, but the fact that they are implemented via proxies is probably noise from the user's perspective.

I wonder if we can get away with TreeNode?

Reply:
I like TreeNode! ...in a vacuum. We'd have need to rename the existing `TreeNode` first. Or be careful with aliasing and imports/exports which might be tricky (or impossible, if they have to both be exported publicly because e.g. API Extractor forces us to).